xcode/Subconscious/Shared/Components/Deck/DeckView.swift

```diff
@@ -0,0 +1,575 @@
+    )
+    
+    case detailStack(DetailStackAction)
+    
+    case setSearchPresented(Bool)
+    case activatedSuggestion(Suggestion)
+    case search(SearchAction)
+    
+    case appear
+    case setDeck([CardModel])
+    
+    case cardPickedUp
+    case cardReleased
+    case cardTapped(CardModel)
+    
+    case chooseCard(CardModel)
+    case skipCard(CardModel)
+    
+    case appendCards([CardModel])
+    case shuffleCardsUpNext([CardModel])
+    
+    case topupDeck
+    case noCardsToDraw
+    
+    case nextCard
+    case cardPresented(CardModel)
+}
+
+extension AppAction {
+    static func from(_ action: DeckAction) -> Self? {
+        switch action {
+        default:
+            return nil
+        }
+    }
+}
+
+extension DeckAction {
+    static func from(_ action: AppAction) -> Self? {
+        switch action {
+        default:
+            return nil
+        }
+    }
+}
+
+typealias DeckEnvironment = AppEnvironment
+
+struct DeckSearchCursor: CursorProtocol {
+    typealias Model = DeckModel
+    typealias ViewModel = SearchModel
+
+    static func get(state: Model) -> ViewModel {
+        state.search
+    }
+
+    static func set(state: Model, inner: ViewModel) -> Model {
+        var model = state
+        model.search = inner
+        return model
+    }
+
+    static func tag(_ action: ViewModel.Action) -> Model.Action {
+        switch action {
+        case .activatedSuggestion(let suggestion):
+            return .activatedSuggestion(suggestion)
+        case .requestPresent(let isPresented):
+            return .setSearchPresented(isPresented)
+        default:
+            return .search(action)
+        }
+    }
+}
+
+struct DeckDetailStackCursor: CursorProtocol {
+    typealias Model = DeckModel
+    typealias ViewModel = DetailStackModel
+
+    static func get(state: Model) -> ViewModel {
+        state.detailStack
+    }
+
+    static func set(state: Model, inner: ViewModel) -> Model {
+        var model = state
+        model.detailStack = inner
+        return model
+    }
+
+    static func tag(_ action: ViewModel.Action) -> Model.Action {
+        switch action {
+        case _:
+            return .detailStack(action)
+        }
+    }
+}
+
+// MARK: Model
+struct DeckModel: ModelProtocol {
+    public static let backlinksToDraw = 1
+    
+    typealias Action = DeckAction
+    typealias Environment = DeckEnvironment
+    
+    static let logger = Logger(
+        subsystem: Config.default.rdns,
+        category: "DeckModel"
+    )
+    
+    var detailStack = DetailStackModel()
+    
+    /// Search HUD
+    var isSearchPresented = false
+    var search = SearchModel(
+        placeholder: "Search or create..."
+    )
+    
+    var deck: Array<CardModel> = []
+    
+    // The set of cards to avoid drawing again, if possible
+    var seen: Set<EntryStub> = []
+    var pointer: Int = 0
+    var author: UserProfile? = nil
+    var selectionFeedback = UISelectionFeedbackGenerator()
```

Comment:
This should be part of the environment. We want models to be dumb data only.

Non-blocking, but let's address in a follow-up.